File: 812-rotate-string/rotate-string.js
/**
 * @param {string} s
 * @param {string} goal
 * @return {boolean}
 */
var rotateString = function(s, goal) {
    if (s.length !== goal.length) {
            return false;
        }
        
        let my = s.split('');
        
        for (let i = 0; i < my.length; i++) {
            if (my.join('') === goal) {
                return true;
            }
            my.push(my.shift());
        }
        
        return false;
};

File: 812-rotate-string/rotate-string.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
         b=s
         my=list(b)
         for i in my:
            if "".join(my) == goal:
                return True 
            my.append(my[0])
            my.remove(my[0])
            
         return False
        
            